rest/src/main/kotlin/builder/channel/thread/StartThreadBuilder.kt

```diff
@@ -0,0 +1,25 @@
+package dev.kord.rest.builder.channel.thread
+
+import dev.kord.common.entity.ArchiveDuration
+import dev.kord.common.entity.ChannelType
+import dev.kord.common.entity.optional.OptionalBoolean
+import dev.kord.common.entity.optional.delegate.delegate
+import dev.kord.common.entity.optional.optional
+import dev.kord.rest.builder.AuditRequestBuilder
+import dev.kord.rest.json.request.StartThreadRequest
+
+class StartThreadBuilder(var name: String, var autoArchiveDuration: ArchiveDuration, var type: ChannelType) : AuditRequestBuilder<StartThreadRequest> {
```

Comment:
make type a `val` here, you don't want to make a switch in the types which makes the builder no longer useful.

Reply:
Only `val` for the `ChannelType`, or for `name` and `autoArchiveDuration` too?

Reply:
`type` only